### PR TITLE
Update OpenEXR and spill out its deps

### PR DESCRIPTION
**Pull request by tad1**
## Description
update OpenEXR to latest
add IMath and Deflate submodule
force OpenEXR to use local deps (IMath & Deflate) instead of using fetchContent

## Testing 
compiles Nabla w/ and w/o `_NBL_COMPILE_WITH_OPEN_EXR_`

## TODO
- [x] make OpenEXR optional dep based on `_NBL_COMPILE_WITH_OPEN_EXR_`
- [x] fix Deflate cmake warrnings; once again make OpenEXR use our Deflate
~~- [ ] remove unnecesary OpenEXR targets~~

### CODE REVIEW

**Comment by tad1** [comment]
currently, IMath & Deflate submodules point to OP's repo, @AnastaZIuk suggested to fork IMath; should Deflate also be forked?

**Comment by AnastaZIuk** [comment]
> currently, IMath & Deflate submodules point to OP's repo, @AnastaZIuk suggested to fork IMath; should Deflate also be forked?

actually as long as we don't plan to modify their CMake we don't have to fork

File: 3rdparty/CMakeLists.txt [hunk 1]
```diff
@@ -222,6 +222,17 @@ unset(CMAKE_INSTALL_MANDIR)
...
+set(LIBDEFLATE_BUILD_GZIP OFF)
+add_subdirectory(libdeflate EXCLUDE_FROM_ALL)
+set(libdeflate_DIR "${CMAKE_CURRENT_BINARY_DIR}/libdeflate")
+
+# Imath
+add_subdirectory(imath EXCLUDE_FROM_ALL)
+
 # OpenEXR
 # note we are adding it anyway regardless _NBL_COMPILE_WITH_OPEN_EXR_, its because we need Half dependecy 
 # it generates on fly in its build directory which doesnt have to be linked (class with inline methods),
```
Comment (tad1) [inline 3rdparty/CMakeLists.txt:238]: since half.h is now part of IMath 3.0 we can disable OpenEXR (and Deflate) when it's not needed

Comment (AnastaZIuk) [reply 3rdparty/CMakeLists.txt:238]: okay go ahead

Comment (tad1) [reply 3rdparty/CMakeLists.txt]: Checked libs `OpenEXRUtil`, `OpenEXRCore`, `Iex`, `IlmThread` (OpenEXR libs); Neither Nabla, nor any other dependency import their headers. I assume it's safe to make OpenEXR optional.

**Pull request merged**